kubectl-plugin/pkg/cmd/scale/scale_cluster_test.go

```diff
@@ -0,0 +1,244 @@
+package scale
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/ray-project/kuberay/kubectl-plugin/pkg/util"
+	"github.com/ray-project/kuberay/kubectl-plugin/pkg/util/client"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	kubefake "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/utils/ptr"
+
+	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+	rayClientFake "github.com/ray-project/kuberay/ray-operator/pkg/client/clientset/versioned/fake"
+)
+
+func TestRayScaleClusterComplete(t *testing.T) {
+	tests := []struct {
+		name              string
+		namespace         string
+		expectedNamespace string
+		args              []string
+	}{
+		{
+			name:              "namespace should be set to 'default' if not specified",
+			args:              []string{"my-workergroup"},
+			expectedNamespace: "default",
+		},
+		{
+			name:              "namespace and worker group should be set correctly",
+			args:              []string{"my-workergroup"},
+			namespace:         "DEADBEEF",
+			expectedNamespace: "DEADBEEF",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			testStreams, _, _, _ := genericclioptions.NewTestIOStreams()
+			fakeScaleClusterOptions := NewScaleClusterOptions(testStreams)
+			if tc.namespace != "" {
+				fakeScaleClusterOptions.configFlags.Namespace = &tc.namespace
+			}
+
+			err := fakeScaleClusterOptions.Complete([]string{"my-workergroup"})
+
+			require.NoError(t, err)
+			assert.Equal(t, tc.expectedNamespace, *fakeScaleClusterOptions.configFlags.Namespace)
+			assert.Equal(t, "my-workergroup", fakeScaleClusterOptions.workerGroup)
+		})
+	}
+}
+
+func TestRayScaleClusterValidate(t *testing.T) {
+	testStreams, _, _, _ := genericclioptions.NewTestIOStreams()
+
+	kubeConfigWithCurrentContext, err := util.CreateTempKubeConfigFile(t, "test-context")
+	require.NoError(t, err)
+
+	kubeConfigWithoutCurrentContext, err := util.CreateTempKubeConfigFile(t, "")
+	require.NoError(t, err)
+
+	tests := []struct {
+		name        string
+		opts        *ScaleClusterOptions
+		expect      string
+		expectError string
+	}{
+		{
+			name: "should error when no context is set",
+			opts: &ScaleClusterOptions{
+				configFlags: &genericclioptions.ConfigFlags{
+					KubeConfig: &kubeConfigWithoutCurrentContext,
+				},
+				ioStreams: &testStreams,
+			},
+			expectError: "no context is currently set, use \"--context\" or \"kubectl config use-context <context>\" to select a new one",
```

Comment:
I remember that this context check test has been moved to another file to avoid duplicates, right?

https://github.com/ray-project/kuberay/blob/349ab01372617b79b5f98cc6063dd3ad7baa5c12/kubectl-plugin/pkg/util/kubeconfig_test.go#L12